CONTRIBUTING.md

```diff
@@ -0,0 +1,856 @@
+
+### Curly braces
+
+Curly braces are required for blocks if the body takes up more than one line. This goes for functions as well. They are optional if it fits within one line, but don't be afraid to use them where you see fit.
+
+Do put an extra space after the first brace and before the final brace in single-line functions.
+
+```js
+// Good
+if (condition) doSomething()
+
+if (condition) {
+	doSomething()
+}
+
+if (condition) {
+	doSomething()
+} else {
+	doSomethingElse()
+}
+
+if (condition) {
+	doSomething()
+	doSomethingElse()
+}
+
+setTimeout(function () { doSomething() }, 0)
+
+setTimeout(function () {
+	doSomething()
+}, 0)
+
+// Bad
+if (condition) { doSomething(); doSomethingElse() }
+
+if (condition)
+	doSomething()
+
+setTimeout(function () { doSomething(); doSomethingElse() }, 0)
+
+setTimeout(function () {doSomething()}, 0) // Body needs some space
+```
+
+### Function declarations
+
+Function declarations should always take more than one line.
+
+```js
+// Good
+function foo() {
+	return bar
+}
+
+// Bad
+function foo() { return bar }
+```
+
+### `if`-`else`
+
+Be consistent with braces in `if`-`else` statements.
+
+```js
+// Good
+if (condition) doSomething()
+else doSomethingElse()
+
+if (condition) {
+	doSomething()
+} else {
+	doSomethingElse()
+}
+
+if (condition) {
+	doSomething()
+} else if (anotherCondition) {
+	doSomethingElse()
+} else {
+	doAnotherThing()
+}
+
+// Okay
+if (condition) doSomething()
+else if (anotherCondition) doSomethingElse()
+else doAnotherThing()
+
+// Bad
+if (condition) doSomething()
+else {
+	doSomethingElse()
+}
+
+if (condition) doSomething()
+else if (anotherCondition) {
+	doSomethingElse()
+} else doAnotherThing()
+
+
+if (condition) {
+	doSomething()
+} else doSomethingElse()
+```
+
+### Empty blocks
+
+Mark empty blocks as intentionally empty via a comment or similar. Don't just leave an empty block and/or semicolon.
+
+```js
+// Good
+for (var i = 0; i < list.length && cond(list[i]); i++) {
+	// empty
+}
+
+// Bad
+for (var i = 0; i < list.length && cond(list[i]); i++) {}
+
+// Also bad
+for (var i = 0; i < list.length && cond(list[i]); i++) {
+	;
+}
+
+// Even worse
+for (var i = 0; i < list.length && cond(list[i]); i++);
+```
+
+### Nesting
+
+Don't nest `if` statements in `else` blocks if you don't need to.
+
+```js
+// Good
+if (test) {
+	doSomething()
+} else if (otherTest) {
+	doSomethingElse()
+}
+
+if (test) {
+	doSomething()
+} else {
+	if (otherTest) {
+		doSomethingElse()
+	}
+
+	doThisOtherThing()
+}
+
+// Bad
+if (test) {
+	doSomething()
+} else {
+	if (otherTest) {
+		doSomethingElse()
+	}
+}
+```
+
+Don't nest curly braces beyond 4 levels deep. This includes blocks, loops, and functions, as well as IIFE wrappers. This also includes the tests. There's rarely a reason to go farther, as most of the time, it's a signal to refactor and/or re-organize your code.
+
+### Brace style
+
+Put the `else`, `catch`, and `finally` on the same line as its closing brace.
```

Comment:
What about `while` in a `do-while` block? (I've try to avoid these since I think they look ugly no matter how you format them, but for consistency it should be mentioned)